www/class/centreonCustomView.class.php

```diff
@@ -839,21 +925,22 @@ public function shareCustomView($params, $userId)
             // share with user groups //
             ////////////////////////////
             $sharedUsergroups = array();
-            $params['lockedUsergroups'] = isset($params['lockedUsergroups']) ? $params['lockedUsergroups'] : array();
-            foreach ($params['lockedUsergroups'] as $lockedUsergroup) {
-                $sharedUsergroups[$lockedUsergroup] = 1;
+            if (!empty($lockedUsergroups)) {
+                foreach ($lockedUsergroups as $lockedUsergroup) {
+                    $sharedUsergroups[$lockedUsergroup] = 1;
+                }
             }
-            $params['unlockedUsergroups'] = isset($params['unlockedUsergroups']) ?
-                $params['unlockedUsergroups'] : array();
-            foreach ($params['unlockedUsergroups'] as $unlockedUsergroup) {
-                $sharedUsergroups[$unlockedUsergroup] = 0;
+            if (!empty($unlockedUsergroups)) {
+                foreach ($unlockedUsergroups as $unlockedUsergroup) {
+                    $sharedUsergroups[$unlockedUsergroup] = 0;
+                }
             }
```

Comment:
Don't need to check if array is empty, it will already avoid loop if empty
```suggestion
            foreach ($unlockedUsergroups as $unlockedUsergroup) {
                $sharedUsergroups[$unlockedUsergroup] = 0;
            }
```